src/openMVG/sfm/pipelines/sfm_regions_provider.hpp

```diff
@@ -42,6 +42,12 @@ struct Regions_Provider
     for (Views::const_iterator iter = sfm_data.GetViews().begin();
       iter != sfm_data.GetViews().end() && bContinue; ++iter)
     {
+      if (my_progress_bar.hasBeenCanceled()) 
+      {
+          bContinue = false;
+          continue;
```

Comment:
Here you use continue, while before you used break.
continue is ok, since you are using the boolean in the loop.
But since there is a multitihread context, I think the bContinue must be in a pragma critical scope